src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -472,20 +472,19 @@ private async Task OnPartialValidationSucceededAsync(ChainedHeader chainedHeader
                     }
                 }
 
-                // Partial validation should continue if:
-                //  1: There are more blocks to validate.
-                //  2: A full validation did not fail (if full validation was needed).
-                bool nextValidationNeeded = chainedHeaderBlocksToValidate != null;
-                bool fullValidationFailed = (connectBlocksResult != null) && !connectBlocksResult.Succeeded;
-                if (nextValidationNeeded && !fullValidationFailed)
+                // If more blocks are available continue validation.
+                if (chainedHeaderBlocksToValidate != null)
                 {
-                    this.logger.LogTrace("Partial validation of {0} block will be started.", chainedHeaderBlocksToValidate.Count);
-
+                    // Validate the next blocks if validation was not needed, or if needed then it succeeded.
+                    if ((connectBlocksResult == null) || connectBlocksResult.Succeeded)
```

Comment:
Can we maybe rename `connectBlocksResult` to `fullValidationResult`? Just so that logically it reads better ;)

Reply:
I agree on changing the name variable if the meaning is that a full validation result happened

Reply:
What about creating an extension for `ValidationResult` like `IsNullOrSucceeded`
this way can be used on any inherited validation results you have and can simplify the checks

e.g.
`if ((connectBlocksResult == null) || connectBlocksResult.Succeeded)`

becomes
`if(connectBlocksResult.IsNullOrSucceeded())`

Reply:
I would not necessary change the param name because this is not only validation its also connecting the block on top of the previous block when we do full validation.